extensions/azurecore/src/account-provider/azureAccountProvider.ts

```diff
@@ -144,17 +146,41 @@ export class AzureAccountProvider implements azdata.AccountProvider, vscode.Disp
 		await this.initCompletePromise;
 		const azureAuth = this.getAuthMethod(account);
 		if (azureAuth) {
+			let identifier = resource.toString() + '|' + tenantId;
 			Logger.piiSanitized(`Getting account security token for ${JSON.stringify(account.key)} (tenant ${tenantId}). Auth Method = ${azureAuth.userFriendlyName}`, [], []);
 			if (this.authLibrary === Constants.AuthLibrary.MSAL) {
+				// Fetch cached token from account properties if token is available and valid.
+				if (account.properties.tokenMap?.[identifier]) {
+					let encryptedAccessToken = account.properties.tokenMap[identifier];
+					try {
+						let accessToken = await this.msalCacheProvider.decryptToken(encryptedAccessToken!);
+						if (this.isValidToken(accessToken)) {
```

Comment:
Consider logging a message if it isn't a valid token for some reason

Reply:
This would happen every hour when tokens expire and we already have logs that capture a new token request - so one more log would not add any value.